### PR TITLE
UCP: call user callback on ucp_ep_close_nbx.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Use user passed params for `ucp_ep_flush_internal` in `ucp_ep_close_nbx` to handle callback and user data fields.

## Why ?
To unify ucp API, and makes easier to implement JUCX logic.

### CODE REVIEW

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -283,7 +283,11 @@ typedef enum {
                                               flag is not set then
                                               @ref ucp_ep_close_nbx schedules
                                               flushes on all outstanding
-                                              operations. */
+                                              operations.
+                                              @note this flag is incompatible
+                                              with a UCP_OP_ATTR_FLAG_NO_IMM_CMPL,
```
Comment (yosefe) [inline src/ucp/api/ucp.h:288]: `with @ref UCP_OP_ATTR_FLAG_NO_IMM_CMPL`

File: test/gtest/ucp/test_ucp_wireup.cc [hunk 2]
```diff
@@ -806,6 +817,32 @@ UCS_TEST_P(test_ucp_wireup_2sided, connect_disconnect) {
     }
 }
 
+UCS_TEST_P(test_ucp_wireup_2sided, close_nbx_callback) {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_wireup.cc:820]: 👍

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@tonycurtis can you pls take a look on small doc change?

**Comment by yosefe** [comment]
bot:pipe:retest

**Review by tonycurtis — CHANGES_REQUESTED** [review]
ucp.h:289: immediate

**Comment by petro-rudenko** [comment]
Fixed

**Comment by petro-rudenko** [comment]
iodemo (No traffic)

/azp run

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**